lua/telescope/actions/set.lua

```diff
@@ -40,6 +39,24 @@ set.select = function(prompt_bufnr, type)
   return set.edit(prompt_bufnr, action_state.select_key_to_edit_key(type))
 end
 
+local edit_buffer
+do
+  local map = {
+    edit = 'buffer',
+    new = 'sbuffer',
+    vnew = 'vert sbuffer',
+    tabedit = 'tab sb',
+  }
+
+  edit_buffer = function(command, bufnr)
+    command = map[command]
+    if command == nil then
+      error('There was no associated buffer command')
+    end
+    vim.cmd(string.format("%s %d", command, bufnr))
+  end
+end
```

Comment:
a version without if else :laughing:, ignore it if you don't like it its okay. I'm just sharing my thoughts

```lua
local errmsg = "There was no associated buffer command"

edit_buffer = function(command, bufnr)
  return map[command] and vim.cmd(map[command]  .. " " .. bufnr) or error(errmsg) 
end

```